cvat/apps/engine/views.py

```diff
@@ -674,78 +716,14 @@ def data(self, request, pk):
-
-                    data_quality = FrameProvider.Quality.COMPRESSED \
-                        if data_quality == 'compressed' else FrameProvider.Quality.ORIGINAL
-
-                    #TODO: av.FFmpegError processing
-                    if settings.USE_CACHE and db_data.storage_method == StorageMethodChoice.CACHE:
-                        buff, mime_type = frame_provider.get_chunk(data_id, data_quality)
-                        return HttpResponse(buff.getvalue(), content_type=mime_type)
-
-                    # Follow symbol links if the chunk is a link on a real image otherwise
-                    # mimetype detection inside sendfile will work incorrectly.
-                    path = os.path.realpath(frame_provider.get_chunk(data_id, data_quality))
-                    return sendfile(request, path)
-
-                elif data_type == 'frame':
-                    data_id = int(data_id)
-                    data_quality = FrameProvider.Quality.COMPRESSED \
-                        if data_quality == 'compressed' else FrameProvider.Quality.ORIGINAL
-                    buf, mime = frame_provider.get_frame(data_id, data_quality)
-
-                    return HttpResponse(buf.getvalue(), content_type=mime)
-
-                elif data_type == 'preview':
-                    return sendfile(request, frame_provider.get_preview())
-
-                elif data_type == 'context_image':
-                    data_id = int(data_id)
-                    image = Image.objects.get(data_id=db_data.id, frame=data_id)
-                    for i in image.related_files.all():
-                        path = os.path.realpath(str(i.path))
-                        image = cv2.imread(path)
-                        success, result = cv2.imencode('.JPEG', image)
-                        if not success:
-                            raise Exception('Failed to encode image to ".jpeg" format')
-                        return HttpResponse(io.BytesIO(result.tobytes()), content_type='image/jpeg')
-                    return Response(data='No context image related to the frame',
-                                    status=status.HTTP_404_NOT_FOUND)
-                else:
-                    return Response(data='unknown data type {}.'.format(data_type), status=status.HTTP_400_BAD_REQUEST)
-            except APIException as e:
-                return Response(data=e.get_full_details(), status=e.status_code)
-            except FileNotFoundError as ex:
-                msg = f"{ex.strerror} {ex.filename}"
-                slogger.task[pk].error(msg, exc_info=True)
-                return Response(data=msg, status=status.HTTP_404_NOT_FOUND)
-            except Exception as e:
-                msg = 'cannot get requested data type: {}, number: {}, quality: {}'.format(data_type, data_id, data_quality)
-                slogger.task[pk].error(msg, exc_info=True)
-                return Response(data=msg + '\n' + str(e), status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Why do we remove the except block?

Reply:
In general the block isn't necessary anymore. Now it throws an exception which can be handled by DRF automatically (e.g. ValidationError). When we refactor the REST API, it will be necessary to check all HTTP error codes from our methods. Also the block isn't good: `except Exception as e:` will catch all possible exceptions. It is a bad idea.

Reply:
@nmanovic, It makes the exception message less informative and user-friendly.
For example, In the case with share data expected something like this:
![image](https://user-images.githubusercontent.com/49038720/146745068-d3875dbd-633a-4181-af97-ddc2892f5483.png)

Reply:
In this case we need to handle a specific exception: https://docs.python.org/3/library/exceptions.html?highlight=filenotfound#FileNotFoundError. I will look at the case.